### DISCUSSION TITLE
Context deadline exceeded

### DISCUSSION THREAD

**Discussion by dberrocal-git** [qdm12/ddns-updater#213] [Q&A]
The update is delayed for minuntes, until it eventually manages to get the IP address, resulting in every call to the address to get the IP, for example:

ERROR Get "https://api.ipify.org": context deadline exceeded

I have changed the option:          HTTP_TIMEOUT=50s

But when I restart the environment it instantly jumps to context deadline exceeded

What could I be doing wrong?

**Comment by qdm12** [comment]
Can you share your logs (hide credentials)? I checked the code (on my phone), it looks fine but I'll recheck tomorrow more in depth.

Also you may want to use DNS as a way to get your IP address with `PUBLICIP_FETCHERS=dns` that may be a workaround for now.

**Comment by qdm12** [comment]
Uh strange, I reviewed the code and it looks like it's applying the http timeout correctly. I'm going to refractor a lot of the configuration / env variables code soon (tomorrow I think) so that might squish out that issue, I'll get back to you.

**Reply by dberrocal-git** [reply]
Hello again, it seems that it has been fixed by setting a specific DNS server in the container (bridge) network, I think it is a matter of the connectivity of the containers that does not resolve names at the beginning (¿) that may explain why it gives the context deadline exceeded - timeout - error.

Thanks for your time, if you need me to try things, reach me.

**Reply by qdm12** [reply]
Strange indeed. I think the container does a connectivity check (trying to dns resole + http get + https get github.com) at start so you should had seen an error there.

Also if it's a DNS issue usually the error message would mention it's a resolve issue, so it might not be that. What exactly did you change? You specified a public DNS in your container definition?

**Reply by dberrocal-git** [reply]
Yes, I also changed the time zone but nothing more
